Sources/GRPCHTTP2TransportNIOTransportServices/TLSConfig.swift

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if canImport(Network)
+@preconcurrency public import Network
+
+@available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+extension HTTP2ServerTransport.TransportServices.Config {
+  /// The security configuration for this connection.
+  public struct TransportSecurity: Sendable {
+    package enum Wrapped: Sendable {
+      case plaintext
+      case tls(TLS)
+    }
+
+    package let wrapped: Wrapped
+
+    /// This connection is plaintext: no encryption will take place.
+    public static let plaintext = Self(wrapped: .plaintext)
+
+    /// This connection will use TLS.
+    public static func tls(_ tls: TLS) -> Self {
+      Self(wrapped: .tls(tls))
+    }
+  }
+
+  public struct TLS: Sendable {
+    /// The `SecIdentity` to be used when setting up TLS.
+    public var identity: SecIdentity
+
+    /// An optional hostname override to use during certificate verification.
+    public var hostnameOverride: String?
+
+    /// Whether ALPN is required.
+    ///
+    /// If this is set to `true` but the client does not support ALPN, then the connection will be rejected.
+    public var requireALPN: Bool
+
+    /// Create a new HTTP2 NIO Transport Services transport TLS config, with some values defaulted:
+    /// - `requireALPN` equals `false`
+    /// - `hostnameOverride` equals `nil`
+    ///
+    /// - Returns: A new HTTP2 NIO Transport Services transport TLS config.
+    public static func defaults(
+      identity: SecIdentity
+    ) -> Self {
+      Self.init(
+        identity: identity,
+        hostnameOverride: nil,
+        requireALPN: false
+      )
+    }
+
+    /// Create a new HTTP2 NIO Transport Services transport TLS config, with some values defaulted to match
+    /// the requirements of mTLS:
+    /// - `requireALPN` equals `false`
+    /// - `hostnameOverride` equals `""`
+    ///
+    /// - Returns: A new HTTP2 NIO Transport Services transport TLS config.
+    public static func mTLS(
+      identity: SecIdentity
+    ) -> Self {
+      Self.init(
```

Comment:
Related to https://github.com/grpc/grpc-swift/pull/2040#discussion_r1747425984 I guess - we have used this method in v1, but it's not clear to me the reason why we're using it is because of mTLS. Is that indeed why? As I said, the docs for these methods never mention mTLS so I'm at a bit of a loss.

Reply:
Decided to not expose these anymore, as configuring mTLS on Network.framework is a lot of work.